### PR TITLE
Lots of minor fixes (typos, code errors, formatting, etc.)

### PR DESCRIPTION
**Pull request by rkawulak**

### CODE REVIEW

**Comment by blakehawkins** [comment]
@rkawulak I think you should revise your commit once more.

**Comment by rkawulak** [comment]
Yeah, apparently my changes are relative to a wrong baseline for some reason... :/

**Comment by kirbyfan64** [comment]
> `using` alias definitiTons

?

**Comment by rkawulak** [comment]
The GitHub's editor is doing some very weird stuff, but hopefully now it's OK.

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -2329,7 +2329,7 @@ In fact, C++98's standard library already used this convenient feature, because
 For example, given a `set<string> myset`, consider:
 
     // C++98
-	result = myset.insert( “Hello” );
```
Comment (rkawulak) [inline CppCoreGuidelines.md]: Note: removed some Unicode whitespace characters not visible in the diff but showing as squares when the doc is rendered.

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -2338,7 +2338,7 @@ With C++11 we can write this, putting the results directly in existing local var
 	Someothertype success;                                  // used these variables for some other purpose
 
 	tie( iter, success ) = myset.insert( “Hello” );         // normal return value
-	if (success) do_something_with( iter );
```
Comment (rkawulak) [inline CppCoreGuidelines.md]: ditto

File: CppCoreGuidelines.md [hunk 3]
```diff
@@ -2599,7 +2599,7 @@ For passthrough functions that pass in parameters (by ordinary reference or by p
 	    stage encryptor  ([] (buffer& b){ encrypt(b); });
 	    stage compressor ([&](buffer& b){ compress(b); encryptor.process(b); });
 	    stage decorator  ([&](buffer& b){ decorate(b); compressor.process(b); });
-	    for (auto& b : bufs) { decorator.process(b); }
```
Comment (rkawulak) [inline CppCoreGuidelines.md]: ditto

File: CppCoreGuidelines.md [hunk 4]
```diff
@@ -6993,17 +6993,17 @@ The definition of `a2` is C but not C++ and is considered a security risk
 
 	widget x;	// should be const, but:
 	for(auto i=2; i <= N; ++i) {             // this could be some
-	    x += some_obj.do_something_with(i);  // arbitrarily long code
-	}                                        // needed to initialize x
```
Comment (rkawulak) [inline CppCoreGuidelines.md]: ditto

File: CppCoreGuidelines.md [hunk 5]
```diff
@@ -6993,17 +6993,17 @@ The definition of `a2` is C but not C++ and is considered a security risk
...
+	    x += some_obj.do_something_with(i);  // arbitrarily long code
+	}                                        // needed to initialize x
 	// from here, x should be const, but we can’t say so in code in this style
 
 **Example; good**:
 
 	const widget x = [&]{
 	    widget val; 		// asume that widget has a default constructor
-		
for(auto i=2; i <= N; ++i) {             // this could be some
```
Comment (rkawulak) [inline CppCoreGuidelines.md]: ditto

File: CppCoreGuidelines.md [hunk 6]
```diff
@@ -6993,17 +6993,17 @@ The definition of `a2` is C but not C++ and is considered a security risk
...
+		for(auto i=2; i <= N; ++i) {             // this could be some
 		    val += some_obj.do_something_with(i);// arbitrarily long code
-		}                                        // needed to initialize x
```
Comment (rkawulak) [inline CppCoreGuidelines.md]: ditto

File: CppCoreGuidelines.md [hunk 7]
```diff
@@ -11589,7 +11589,7 @@ Some conventions capitalize the first letter, some don't.
 **Note**: Try to be consistent in your use of acronyms, lengths of identifiers:
 
 	int mtbf {12};
-	int mean_time_between_failor {12};		// make up your mind
+	int mean_time_between_failure {12};		// make up your mind
```
Comment (justanotheranonymoususer) [inline CppCoreGuidelines.md]: I think that the typo here was intended.

**Pull request closed without merging**